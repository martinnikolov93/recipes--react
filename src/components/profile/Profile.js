import React, { useContext } from 'react';
import UserContext from '../../Context';
import RecipesWrapper from '../recipes-wrapper/RecipesWrapper';
import withFetching from '../../hocs/withFetching'

const Profile = () => {
    const context = useContext(UserContext)
    const RecipesWrapperWithFetch = withFetching(`/user/recipes?id=${context.user._id}`)(RecipesWrapper)
    return (
        <>
            <h3>Email: {context.user.email}</h3>
            <h3>Your recipes:</h3>
            <RecipesWrapperWithFetch />
        </>
    )
}

export default Profile