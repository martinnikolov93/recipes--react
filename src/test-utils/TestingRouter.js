import React from 'react';
import UserContext from '../Context';
import { BrowserRouter, Route } from 'react-router-dom'

const TestingRouter = ({user, children}) => {
    return (
        <BrowserRouter>
            <Route>
                <UserContext.Provider value={user}>
                    {children}
                </UserContext.Provider>
            </Route>
        </BrowserRouter>
    )
}

export default TestingRouter