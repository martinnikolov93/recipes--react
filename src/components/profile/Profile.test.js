import React from 'react';
import renderer from 'react-test-renderer'
import Profile from './Profile';
import TestingRouter from '../../test-utils/TestingRouter';

jest.mock('../../hocs/withFetching.js', () => {
    return () => () => {
        return 'With Fetching Hoc'
    }
})

const userDummy = {
    loggedIn: true,
    user: {
        email: 'test@test.com'
    }
}

describe('Profile component', () => {
    it('Should render profile component', () => {
        const tree = renderer.create(
            <TestingRouter user={userDummy}>
                <Profile />
            </TestingRouter>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})