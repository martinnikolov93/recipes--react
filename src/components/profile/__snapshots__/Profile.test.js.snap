// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component Should render profile component 1`] = `
Array [
  <h3>
    Email: 
    test@test.com
  </h3>,
  <h3>
    Your recipes:
  </h3>,
  <With Fetching Hoc />,
]
`;
